pkgs/shells/xonsh/wrapper.nix

```
@@ -0,0 +1,24 @@
+{ runCommand
+, xonsh-unwrapped
+, lib
+, extraPackages ? (ps: [ ])
+}:
+
+let
+  xonsh = xonsh-unwrapped;
+  inherit (xonsh.passthru) python;
+
+  pythonEnv = python.withPackages (ps: [
+    (ps.toPythonModule xonsh)
+  ] ++ extraPackages ps);
+
+in
+runCommand "${xonsh.pname}-${xonsh.version}"
+{
+  inherit (xonsh) pname version meta;
+} ''
+  mkdir -p $out/bin
+  for bin in ${lib.getBin xonsh}/bin/*; do
+    ln -s ${pythonEnv}/bin/$(basename "$bin") $out/bin/
+  done
```

```suggestion
  lndir ${lib.getBin xonsh}/bin $out/bin
```
lndir is under xorg.lndir but not important just an idea

---

If you look a bit closer you can see that these are not the same.
What I do in this wrapper is to link a subset of binaries (those available in the xonsh package) from a python environment.
This is a filtering mechanism.